### ISSUE TITLE
Emergency calls (110 or 112) are not working (well)

### ISSUE THREAD

**Issue by gordon-fr33man** [FossifyOrg/Phone#250]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Phone/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Phone/issues) or [closed](https://github.com/FossifyOrg/Phone/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Phone/discussions) or [closed](https://github.com/FossifyOrg/Phone/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Phone/blob/master/CONTRIBUTING.md).

### Affected app version

1.2.0

### Affected Android/Custom ROM version

Android 13

### Affected device model

Fairphone 4

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Start an emergency call to 110 or 112 (please don't, if it's not a real emergency!)

### Expected behavior

An emergency call to 110 (police in Germany) or 112 (European emergency call) should work just like any other call.

### Actual behavior

When I dial one of the two emergency numbers (110 [German police] oder 112 [european emergency call]) and start the call, nothing happens according to the app. A call is started in the background, but the normal screen that normally appears for every ongoing call does not appear. Instead, I continue to see the numpad and the digits I have just typed and dialled have disappeared. Even afterwards, there is no emergency call listed in the call list.

### Additional information

This is my first time using GitHub, please be kind :D

**Comment by jimbo-the-prince** [comment]
I have also experienced this issue when calling either 911 or using the 'emergency call' lock screen button to place a call to 911; however I also had issues just making the call itself, there was some sort of deadly popup in the way I have forgotten details about and I had to 'confirm' calling them. This is a literally deadly bug and needs to be taken very seriously.

**Comment by XandaPanda42** [comment]
Got hit with this bug yesterday. The call was placed correctly, and both parties could hear, but the UI had disappeared and I couldn't access the call's page to press keys or hang up. At the end of the call, I had to ask the operator to hang up which was awkward.

I might be wrong, but I remember reading something about third party apps not being able to use the Call API for emergency numbers. If a third party app requests that the API dial an emergency number, the stock phone app is meant to take over. 

However, since most of the users of Fossify Phone are attempting to get away from the stock phone apps, a lot of us have them disabled. This means that while the call API does still work to dial the number, the stock app (Google Phone in my case) was unable to open to manage the call, and Fossify can request that the API make a call, but the emergency override means that the call is sent to the stock app, so Fossify can't see it, or access or hang up that call.

I can't remember where I read it or how accurate it is, but it's a start.

Also of note: The class `android.telephony.emergency` does not appear in the source anywhere, so even if the rules had changed and it was allowed now, the app couldn't do it anyway. It's not so much of a bug as a missing CRITICAL feature.

I'm almost guaranteed to have to call an emergency number again in the next month or two, so if there's anyone involved with the project who can tell me the best way to get more detailed logs of these events, let me know. Since Foss can't access emergency calls, there's nothing in the Call History page.

**Comment by XandaPanda42** [comment]
@naveensingh,

Found some more info to clarify, the class that helps manage emergency dialing can be found [here.](https://developer.android.com/reference/android/telephony/emergency/EmergencyNumber)

It looks like it was added in Android 10, so my previous comment is out of date. Third party apps should be able to handle emergency dialing for any version later than Android 10 (API level 29).

I understand that this kind of issue is difficult to address given the nature of emergency calling, so if there's anything I can do to assist, please let me know. As I said, I'll likely need to call again at some point, so if there's a way I can enable more verbose logging to assist with testing, if you ever build a version you need tested, I can have the APK installed on my old spare phone and have it at the ready just in case.

There is also another alternative if you'd be interested. In Aus at least, people belonging to small businesses in the tech industry can register for approval to test their systems to handle emergency numbers. I'd be happy to email and ask some questions. 

I'd like to help, but I've never been great with Java or Kotlin so there's little I can do in terms of code contributions, but I already fit most of the criteria for testing approval. And since Apple and Google pretty much forced Emergency Services to standardize across the world, if it works in Aus, it'll possibly make it work in other places too. 

Let me know what you think :-)

**Comment by naveensingh** [comment]
@gordon-fr33man @jimbo-the-prince @Hoerli1337, @smowtenshi

Do you guys also have the stock dialer/phone app disabled?

**Comment by naveensingh** [comment]
From the [documentation](https://developer.android.com/develop/connectivity/telecom/dialer-app#becoming_the_default_phone_app): 

> **The preloaded dialer will ALWAYS be used when the user places an emergency call**, even if your app fills the [RoleManager.ROLE_DIALER](https://developer.android.com/reference/android/app/role/RoleManager#ROLE_DIALER) role. To ensure an optimal experience when placing an emergency call, the default dialer should ALWAYS use [TelecomManager.placeCall(Uri, Bundle)](https://developer.android.com/reference/android/telecom/TelecomManager#placeCall(android.net.Uri,%20android.os.Bundle)) to place calls (including emergency calls). This ensures that the platform is able to verify that the request came from the default dialer. If a non-preloaded dialer app uses [Intent#ACTION_CALL](https://developer.android.com/reference/android/content/Intent#ACTION_CALL) to place an emergency call, it will be raised to the preloaded dialer app using [Intent#ACTION_DIAL](https://developer.android.com/reference/android/content/Intent#ACTION_DIAL) for confirmation; this is a suboptimal user experience.

So there isn't much we can do here, maybe just show a warning when the the system dialer app has been disabled.

Related:
 - https://gitlab.com/LineageOS/issues/android/-/issues/8026
 - https://www.reddit.com/r/LineageOS/comments/ujh5tx/3rd_party_dialer_app_which_supports_emergency/

**Comment by gordon-fr33man** [comment]
> @gordon-fr33man @jimbo-the-prince @Hoerli1337, @smowtenshi
> 
> Do you guys also have the stock dialer/phone app disabled?

Yes, I have. But I think, reading this, I will reactivate it at least with limited rights ...

**Comment by naveensingh** [comment]
> show a warning when the the system dialer app has been disabled

That may not always work so I have added a FAQ entry instead.

**Comment by WebworkrNet** [comment]
@naveensingh 

Motorola moto g34 5G
Android 14

I'm not sure whether I should let a person with limited technical experience with smartphones in general and no experience with Android in particular use the dialer. In an emergency, the person should not have to deal with any technical restrictions or problems. 

**If the standard dialer remains active in the background, can it really be guaranteed that such a call will work properly via Fossify in an emergency?
Will the emergency call work properly via Fossify or alternatively be forwarded to the standard dialer?**

Android requires a default app to be set for the dialer. **Would the user have to change it before making an emergency call?** I would classify this as excessive demands due to the above-mentioned circumstances.

Even I, who have a professional background in IT, have doubts as to whether I would be able to react appropriately in an emergency situation. I also wonder whether I would want to deal with such a situation at all.
In the worst case, it could be a matter of life and death.

**Comment by naveensingh** [comment]
> If the standard dialer remains active in the background, can it really be guaranteed that such a call will work properly via Fossify in an emergency?

Anything other than the default setup provided by a vendor has risks, so I wouldn't exactly say **guaranteed** but as per my testing on different phones, I think emergency calls will work without a hitch as long as you do not disable or restrict your preloaded/system dialer in any way.

> Android requires a default app to be set for the dialer. Would the user have to change it before making an emergency call?

No, you do not need to change the default dialer before making the call. Fossify Phone places emergency calls like any other, and the rest is done by the system. The whole point of this [unconditional switch](https://github.com/FossifyOrg/Phone/issues/250#issuecomment-2585203666) to the system dialer is to ensure reliability during an emergency.
